Django/BookLibrary/library/book_details.py

```diff
@@ -0,0 +1,105 @@
+from bs4 import BeautifulSoup
+import requests
+import time
+
+''' This Class will return the Book Details such as 
+"Cover Image, Amazon Buy Link, Flipkart Buy Link, Good Reads rating" '''
+class BookDetails:
+    
+    #Initializing the urls for the methods
+    def __init__(self, search_query):
+        self.flipkart_url = f'https://www.flipkart.com/search?q={search_query} Book'
+        self.amazon_url = f'https://www.amazon.com/s?k={search_query} Book'
+        self.good_reads_url = f'https://www.goodreads.com/search?q={search_query}'
+        self.headers = {
+            "User-Agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:66.0) Gecko/20100101 Firefox/66.0",
+            "Accept-Encoding":"gzip, deflate", "Accept":"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
+            "DNT":"1","Connection":"close", "Upgrade-Insecure-Requests":"1"
+        }
+
+    #This function returns Cover Image of the Book
+    def cover_image(self):     
+        try:
+            good_reads_response = requests.get(self.good_reads_url, headers=self.headers)
+            if not good_reads_response:
+                return "https://www.goodreads.com/not-found"
```

Comment:
I see you are use www.goodreads.com in multiple places. You can assign it to a constant value and re-use it.